<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>adminw</class>
 <widget class="QWidget" name="adminw">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>746</width>
    <height>584</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Admin/moderator functions</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <property name="sizeConstraint">
        <enum>QLayout::SetMaximumSize</enum>
       </property>
       <item>
        <widget class="QPushButton" name="pushButton">
         <property name="text">
          <string>Refresh</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="subgod_tab">
        <attribute name="title">
         <string>minions and blocks</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <layout class="QHBoxLayout" name="_2">
           <property name="spacing">
            <number>6</number>
           </property>
           <property name="margin">
            <number>0</number>
           </property>
           <item>
            <layout class="QVBoxLayout" name="_3">
             <property name="spacing">
              <number>6</number>
             </property>
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <layout class="QHBoxLayout" name="_4">
               <property name="spacing">
                <number>6</number>
               </property>
               <property name="margin">
                <number>0</number>
               </property>
               <item>
                <widget class="QLabel" name="label_4">
                 <property name="text">
                  <string>UID</string>
                 </property>
                 <property name="buddy">
                  <cstring>uid_edit</cstring>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="uid_edit"/>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="_5">
               <property name="spacing">
                <number>6</number>
               </property>
               <property name="margin">
                <number>0</number>
               </property>
               <item>
                <widget class="QLabel" name="label_5">
                 <property name="text">
                  <string>Why</string>
                 </property>
                 <property name="buddy">
                  <cstring>why_edit</cstring>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="why_edit"/>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QGridLayout" name="_6">
             <property name="margin">
              <number>0</number>
             </property>
             <property name="spacing">
              <number>6</number>
             </property>
             <item row="1" column="2">
              <widget class="QSpinBox" name="minutes_spin"/>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Days</string>
               </property>
               <property name="buddy">
                <cstring>days_spin</cstring>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="hours_spin"/>
             </item>
             <item row="1" column="0">
              <widget class="QSpinBox" name="days_spin"/>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Hours</string>
               </property>
               <property name="buddy">
                <cstring>hours_spin</cstring>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Minutes</string>
               </property>
               <property name="buddy">
                <cstring>minutes_spin</cstring>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="_7">
             <property name="spacing">
              <number>6</number>
             </property>
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <widget class="QPushButton" name="block_button">
               <property name="text">
                <string>Block</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="unblock_button">
               <property name="text">
                <string>Unblock</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item row="1" column="0">
          <widget class="QTableWidget" name="block_table">
           <property name="alternatingRowColors">
            <bool>true</bool>
           </property>
           <property name="columnCount">
            <number>7</number>
           </property>
           <column/>
           <column/>
           <column/>
           <column/>
           <column/>
           <column/>
           <column/>
          </widget>
         </item>
         <item row="2" column="0">
          <layout class="QHBoxLayout" name="_8">
           <property name="spacing">
            <number>6</number>
           </property>
           <property name="margin">
            <number>0</number>
           </property>
           <item>
            <widget class="QTableWidget" name="mod_table">
             <property name="columnCount">
              <number>2</number>
             </property>
             <column/>
             <column/>
            </widget>
           </item>
           <item>
            <layout class="QVBoxLayout" name="_9">
             <property name="spacing">
              <number>6</number>
             </property>
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <layout class="QGridLayout" name="_10">
               <property name="margin">
                <number>0</number>
               </property>
               <property name="spacing">
                <number>6</number>
               </property>
               <item row="0" column="0">
                <widget class="QPushButton" name="mod_button">
                 <property name="text">
                  <string>Mod UID</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QPushButton" name="unmod_button">
                 <property name="text">
                  <string>Revoke Mod</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QPushButton" name="clear_mods_button">
                 <property name="text">
                  <string>Clear Mods</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <spacer>
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item row="3" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Server rules</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="update_rules">
             <property name="text">
              <string>Update rules</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item row="4" column="0">
          <widget class="QTextEdit" name="rules"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="lobbies_tab">
        <attribute name="title">
         <string>lobbies</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <layout class="QHBoxLayout" name="_11">
             <item>
              <layout class="QHBoxLayout" name="_12">
               <item>
                <widget class="QLabel" name="label_7">
                 <property name="text">
                  <string>category</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="category_edit"/>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="_13">
               <item>
                <widget class="QLabel" name="label_8">
                 <property name="text">
                  <string>max users</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="max_users_edit"/>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="_14">
               <item>
                <widget class="QLabel" name="label_6">
                 <property name="text">
                  <string>dispname</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="dispname_edit"/>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="_15">
               <item>
                <widget class="QLabel" name="label_9">
                 <property name="text">
                  <string>pw</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="pw_edit"/>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="_16">
             <item>
              <widget class="QPushButton" name="create_user_lobby_button">
               <property name="text">
                <string>Create Lobby</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="remove_user_lobby_button">
               <property name="text">
                <string>Remove Lobby</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item row="1" column="0">
          <widget class="QTableWidget" name="server_table">
           <property name="alternatingRowColors">
            <bool>true</bool>
           </property>
           <property name="selectionMode">
            <enum>QAbstractItemView::NoSelection</enum>
           </property>
           <column>
            <property name="text">
             <string>Host</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>IP</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Port</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Name</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Category</string>
            </property>
           </column>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QTableWidget" name="ulob_table">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="alternatingRowColors">
            <bool>true</bool>
           </property>
           <property name="selectionMode">
            <enum>QAbstractItemView::SingleSelection</enum>
           </property>
           <property name="selectionBehavior">
            <enum>QAbstractItemView::SelectRows</enum>
           </property>
           <column>
            <property name="text">
             <string>id</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>host</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>ip</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>port</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>rating</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>dispname</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>category</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>max</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>subgod</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="sysattr_tab">
        <attribute name="title">
         <string>sysattr</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout">
               <item>
                <widget class="QPushButton" name="set_sys_attr">
                 <property name="text">
                  <string>set sys attr</string>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout">
                 <item>
                  <widget class="QLineEdit" name="sys_name"/>
                 </item>
                 <item>
                  <widget class="QComboBox" name="type_comboBox">
                   <item>
                    <property name="text">
                     <string>nil</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>str</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>int</string>
                    </property>
                   </item>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <widget class="QTextEdit" name="sys_val"/>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QTableWidget" name="sys_table">
               <column>
                <property name="text">
                 <string>Type</string>
                </property>
               </column>
               <column>
                <property name="text">
                 <string>Name</string>
                </property>
               </column>
               <column>
                <property name="text">
                 <string>Val</string>
                </property>
               </column>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
